src/Native/Unix/System.Native/pal_process.c

```diff
@@ -336,10 +405,12 @@ done:;
         *childPid = -1;
 
         errno = priorErrno;
-        return -1;
     }
 
-    return 0;
+    // Restore thread cancel state
+    //pthread_setcancelstate(thread_cancel_state, &thread_cancel_state);
```

Comment:
This should not be commented out, right?

Reply:
My mistake.